src/hooks/useFetch.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+const fetchUrl = (url, after, count, limit = 100, time = 'year') => {
```

Comment:
I think it would be easier to get the `subreddit` as a parameter instead of the `url`. Then you could build the complete URL with this function.

Btw: `fetchUrl` is a bit misleading. It sounds as if this function would fetch something. Remember the [lesson about variable naming](https://www.notion.so/A-single-rule-to-name-your-variables-2733517f45414829b09a071386e1c375)? Maybe it's a good exercise to use that rule here :smile: 

PS: I'll add comments where I think the naming could be improved at other places as well

Reply:
One of the initial problems I ran into was that I used subreddit as the useEffect dependency in useFetch and because it changed every time I started typing in the input box, it caused issues.  So I ended up creating the URL after the subreddit was set and that seemed to fix the problem.  I agree that passing in the subreddit makes more sense.  As I go through your comments, I will see how I can change this up.

Reply:
Right, that makes sense. You could use the `subredditQuery` variable instead. Or you could even use the `useParams` inside your custom hook...

Reply:
subredditQuery works! I don't know why I didn't try that 😑

Reply:
Happens :smile: